model/src/test/kotlin/utils/ExtensionsTest.kt

```diff
@@ -0,0 +1,188 @@
+/*
+ * Copyright (C) 2021 Bosch.IO GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.ossreviewtoolkit.model.utils
+
+import io.kotest.core.spec.style.WordSpec
+import io.kotest.matchers.collections.shouldContain
+import io.kotest.matchers.collections.shouldHaveSize
+import io.kotest.matchers.collections.shouldNotContain
+
+import org.ossreviewtoolkit.model.AnalyzerResult
+import org.ossreviewtoolkit.model.AnalyzerRun
+import org.ossreviewtoolkit.model.Identifier
+import org.ossreviewtoolkit.model.OrtResult
+import org.ossreviewtoolkit.model.Package
+import org.ossreviewtoolkit.model.Repository
+import org.ossreviewtoolkit.model.config.AnalyzerConfiguration
+import org.ossreviewtoolkit.model.config.LicenseChoices
+import org.ossreviewtoolkit.model.config.PackageLicenseChoice
+import org.ossreviewtoolkit.model.config.RepositoryConfiguration
+import org.ossreviewtoolkit.model.licenses.LicenseCategorization
+import org.ossreviewtoolkit.spdx.SpdxSingleLicenseExpression
+import org.ossreviewtoolkit.spdx.model.LicenseChoice
+import org.ossreviewtoolkit.spdx.toSpdx
+import org.ossreviewtoolkit.utils.Environment
+import org.ossreviewtoolkit.utils.ProcessedDeclaredLicense
+
+class ExtensionsTest : WordSpec() {
+    init {
+        "List<Package>.filterByCategories" should {
+            "return only packages from the specified categories" {
+                val packageA = Package.EMPTY.copy(
+                    id = Identifier("Maven:org.ossreviewtoolkit:a:1.0.0"),
+                    declaredLicenses = sortedSetOf("license-a"),
+                    declaredLicensesProcessed = ProcessedDeclaredLicense("license-a".toSpdx())
```

Comment:
I think you could omit setting `declaredLicensesProcessed`, maybe you would have to use "LicenseRef-a" then to make it SPDX conform.

Reply:
I don't think that this would work because `Package.Empty` sets `ProcessedDeclaredLicense.EMPTY` explicitly and it won't be overwritten when using `copy()`.

Reply:
No longer required due to https://github.com/oss-review-toolkit/ort/pull/3935#discussion_r626502111